website/files/models/base.py

```diff
@@ -91,6 +91,19 @@ def restore(self, recursive=True, parent=None):
         TrashedFileNode.remove_one(self)
         return restored
 
+    def get_guid(self, create=False):
+        """Attempt to find a Guid that points to this object.
+        One will be created if requested.
+        :rtype: Guid
+        """
+        try:
+            # Note sometimes multiple GUIDs can exist for
+            # a single object. Just go with the first one
+            return Guid.find(Q('referent', 'eq', self))[0]
```

Comment:
Having trouble with this.
I switched...         return Guid.find(Q('referent', 'eq', self))[0]
to...                     return Guid.find_one(Q('referent', 'eq', self))
but that returns "None"
In the examples I looked at the object of find_one were either "cls" or "FileNode.resolve_class(self.provider, kind)"
Do I need to use the resolve_class method? If so it looks like I would need to introduce a numerical "kind" to the code.